src/main/java/se/kth/sda/skeleton/posts/PostController.java

```diff
@@ -1,8 +1,57 @@
 package se.kth.sda.skeleton.posts;
 
-/*
-    @TODO create the methods needed to implement the API.
-    Don't forget to add necessary annotations.
- */
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import se.kth.sda.skeleton.posts.exeption.*;
+import se.kth.sda.skeleton.posts.PostService;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RequestMapping("/posts")
+@RestController
 public class PostController {
+    PostRepository postRepository;
+    PostService postService;
+
+    @Autowired
+    public PostController(PostRepository postRepository, PostService postService) {
+        this.postRepository = postRepository;
+        this.postService = postService;
+    }
+    /* create a new post */
+    @PostMapping("/")
+    public ResponseEntity<Post> createPost(@Valid @RequestBody Post post) {
+        return ResponseEntity.status(HttpStatus.CREATED)
+                .body(postService.createPost(post));
+    }
```

Comment:
I find the 2 methods with same name:  `createPost()`  confusing ( line 26 and 28)

Reply:
Ok, we are now changing the method names in PostController to match the ones in PostService